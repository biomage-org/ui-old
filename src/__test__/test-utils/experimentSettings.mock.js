import initialState from 'redux/reducers/experimentSettings/initialState';

const sampleifiedConfig = (sampleIds, configToReplicate) => {
  const result = sampleIds.reduce(
    (acum, sampleId) => {
      // eslint-disable-next-line no-param-reassign
      acum[sampleId] = configToReplicate;
      return acum;
    },
    {},
  );

  return result;
};

const generateProcessingConfigMock = (sampleIds) => ({
  classifier: {
    enabled: true,
    prefiltred: false,
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        FDR: 0.1,
      },
    }),
  },
  cellSizeDistribution: {
    enabled: true,
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        minCellSize: 10800,
        binStep: 200,
      },
    }),
  },
  mitochondrialContent: {
    enabled: true,
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        method: 'absoluteThreshold',
        methodSettings: {
          absoluteThreshold: {
            maxFraction: 0.1,
            binStep: 200,
          },
        },
      },
    }),
  },
  numGenesVsNumUmis: {
    enabled: true,
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        regressionType: 'gam',
        regressionTypeSettings: {
          gam: {
            'p.level': 0.00009,
          },
        },
      },
    }),
  },
  doubletScores: {
    enabled: true,
    ...sampleifiedConfig(sampleIds, {
      auto: true,
      enabled: true,
      filterSettings: {
        probabilityThreshold: 0.2,
        binStep: 0.05,
      },
    }),
  },
  dataIntegration: {
    enabled: true,
    dataIntegration: {
      method: 'seuratv4',
      methodSettings: {
        seuratv4: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
        fastmnn: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
        unisample: {
          numGenes: 2000,
          normalisation: 'logNormalize',
        },
      },
    },
    dimensionalityReduction: {
      method: 'rpca',
      numPCs: 30,
      excludeGeneCategories: [],
    },
  },
  configureEmbedding: {
    enabled: true,
    embeddingSettings: {
      method: 'umap',
      methodSettings: {
        umap: {
          minimumDistance: 0.1,
          distanceMetric: 'euclidean',
        },
        tsne: {
          perplexity: 30,
          learningRate: 200,
        },
      },
    },
    clusteringSettings: {
      method: 'louvain',
      methodSettings: {
        louvain: {
          resolution: 0.5,
        },
      },
    },
  },
});

const generateExperimentSettingsMock = (sampleIds) => {
  const mockedProcessingConfig = {
    ...initialState.processing,
    ...generateProcessingConfigMock(sampleIds),
  };

  return {
    ...initialState,
    processing: mockedProcessingConfig,
    originalProcessing: mockedProcessingConfig,
    info: {
      ...initialState.info,
      pipelineVersion: 1,
    },
  };
};

export default generateExperimentSettingsMock;
