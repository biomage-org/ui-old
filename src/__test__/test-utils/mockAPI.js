import _ from 'lodash';

// Default platform data
import cellSetsData from '__test__/data/cell_sets.json';
import backendStatusData from '__test__/data/backend_status.json';
import processingConfigData from '__test__/data/processing_config.json';
import mockDemoExperiments from '__test__/test-utils/mockData/mockDemoExperiments.json';

// A ticket has been created to address this : https://biomage.atlassian.net/browse/BIOMAGE-1553
import {
  responseData,
} from '__test__/test-utils/mockData';

const promiseResponse = (
  response,
  options = {},
) => Promise.resolve(new Response(response, options));

const statusResponse = (code, body) => Promise.resolve({
  status: code,
  body: JSON.stringify(body),
});

const delayedResponse = (response, delay = 10000, options = {}) => new Promise((resolve) => {
  setTimeout(() => {
    resolve(new Response(response, options));
  }, delay);
});

const workerResponse = (body) => promiseResponse(body);

const generateDefaultMockAPIResponses = (experimentId) => ({
  [`experiments/${experimentId}`]: () => promiseResponse(
    JSON.stringify(responseData.experiments.find(({ id }) => id === experimentId)),
  ),
  [`experiments/${experimentId}/processingConfig`]: () => promiseResponse(
    JSON.stringify(processingConfigData),
  ),
  [`experiments/${experimentId}/cellSets`]: () => promiseResponse(
    JSON.stringify(cellSetsData),
  ),
  [`experiments/${experimentId}/backendStatus`]: () => promiseResponse(
    JSON.stringify(backendStatusData),
  ),
  experiments: () => promiseResponse(
    JSON.stringify(responseData.experiments),
  ),
  [`experiments/${experimentId}/samples`]: () => promiseResponse(
    JSON.stringify(responseData.samples[0]),
  ),
  '/v2/experiments/examples': () => promiseResponse(
    JSON.stringify(mockDemoExperiments),
  ),
});

const mockAPI = (apiMapping) => (req) => {
  const path = req.url;

  const key = _.find(
    Object.keys(apiMapping),
    (urlStub) => path.endsWith(urlStub),
  );

  if (!key) {
    return statusResponse({
      status: 404,
      body: `Path ${path} is undefined`,
    });
  }

  return apiMapping[key](req);
};

export default mockAPI;
export {
  generateDefaultMockAPIResponses,
  promiseResponse,
  statusResponse,
  workerResponse,
  delayedResponse,
};
