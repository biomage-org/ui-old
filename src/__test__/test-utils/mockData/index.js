import _ from 'lodash';
import generateMockSamples from './generateMockSamples';
import generateMockExperiments from './generateMockExperiments';
import generateMockProcessingConfig from './generateMockProcessingConfig';

// This file is a work in progress to generate mock data for the endpoints of the test.
// The generated data should be usable as response data to be sent via the mocked API in mockAPI.js
// The module should also expose the generated data that can be imported and used easily in tests
// A ticket has been created to give more thought into this https://biomage.atlassian.net/browse/BIOMAGE-1553

const responseData = {};

responseData.experiments = generateMockExperiments(2);

responseData.samples = [generateMockSamples(
  responseData.experiments[0],
  3,
)];

responseData.processingConfig = generateMockProcessingConfig(3);

// Add samples to first experiment
const samples = responseData.samples[0];
const sampleIds = _.map(samples, 'id');
responseData.experiments[0].samplesOrder = sampleIds;

const { experiments, processingConfig } = responseData;

export {
  experiments,
  samples,
  processingConfig,
  responseData,
};
