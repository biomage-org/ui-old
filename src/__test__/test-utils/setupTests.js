import '@testing-library/jest-dom/extend-expect';
import preloadAll from 'jest-next-dynamic';
import Enzyme from 'enzyme';
import Adapter from '@wojtekmaj/enzyme-adapter-react-17';

Enzyme.configure({ adapter: new Adapter() });

jest.mock('localforage');
jest.mock('utils/pushNotificationMessage');

beforeAll(async () => {
  // Add stuff that needs to run once, before all tests
  await preloadAll();
});

beforeEach(async () => {
  // Add stuff that needs to run before each test
});
