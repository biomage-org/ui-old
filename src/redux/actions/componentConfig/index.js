import loadPlotConfig from 'redux/actions/componentConfig/loadPlotConfig';
import loadComponentConfig from 'redux/actions/componentConfig/loadComponentConfig';
import updatePlotConfig from 'redux/actions/componentConfig/updatePlotConfig';
import savePlotConfig from 'redux/actions/componentConfig/savePlotConfig';
import getDotPlot from 'redux/actions/componentConfig/getDotPlot';
import updatePlotData from 'redux/actions/componentConfig/updatePlotData';
import resetPlotConfig from 'redux/actions/componentConfig/resetPlotConfig';

export {
  loadPlotConfig,
  loadComponentConfig,
  updatePlotConfig,
  savePlotConfig,
  getDotPlot,
  updatePlotData,
  resetPlotConfig,
};
