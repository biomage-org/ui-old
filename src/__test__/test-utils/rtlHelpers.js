import { fireEvent, screen } from '@testing-library/dom';
import userEvent from '@testing-library/user-event';
import { act } from 'react-dom/test-utils';

const selectOption = async (optionName, select) => {
  await userEvent.click(select);
  // userEvent click doesnt trigger the onChange in select
  fireEvent.click(screen.getByTitle(optionName));
};

// dropFilesIntoDropzone takes an input element and simulates droping files into it
// inputElement: this needs to be the actual input element not the dropzone itself, e.g.:
// * <input data-testid='drop-input'/> => screen.getByTestId('drop-input');
// files should be a list of files, e.g.:
// * new File(['content'], 'file.txt', { type: 'text/plain' });
const dropFilesIntoDropzone = async (inputElement, files) => {
  await act(async () => {
    fireEvent.drop(inputElement, {
      target: { files },
    });
  });
};

export {
  // eslint-disable-next-line import/prefer-default-export
  selectOption,
  dropFilesIntoDropzone,
};
