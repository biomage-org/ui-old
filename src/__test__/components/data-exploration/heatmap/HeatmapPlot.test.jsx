import _ from 'lodash';
import React from 'react';
import preloadAll from 'jest-next-dynamic';

import { render, screen, waitFor } from '@testing-library/react';
import { act } from 'react-dom/test-utils';

import { dispatchWorkRequest, seekFromS3 } from 'utils/work/seekWorkResponse';

import { Provider } from 'react-redux';
import fetchMock, { enableFetchMocks } from 'jest-fetch-mock';

import markerGenesData2 from '__test__/data/marker_genes_2.json';
import markerGenesData5 from '__test__/data/marker_genes_5.json';
import cellSetsData from '__test__/data/cell_sets.json';

import { makeStore } from 'redux/store';

import mockAPI, {
  generateDefaultMockAPIResponses,
  delayedResponse,
} from '__test__/test-utils/mockAPI';

import HeatmapPlot from 'components/data-exploration/heatmap/HeatmapPlot';

import { loadProcessingSettings } from 'redux/actions/experimentSettings';
import { loadGeneExpression } from 'redux/actions/genes';

import { loadBackendStatus } from 'redux/actions/backendStatus';

import fake from '__test__/test-utils/constants';
import { loadCellSets, setCellSetHiddenStatus } from 'redux/actions/cellSets';
import { isSubset } from 'utils/arrayUtils';
import { updatePlotConfig } from 'redux/actions/componentConfig';

const experimentId = fake.EXPERIMENT_ID;

// Mock hash so we can control the ETag that is produced by hash.MD5 when fetching work requests
// EtagParams is the object that's passed to the function which generates ETag in fetchWork
jest.mock('object-hash', () => {
  const objectHash = jest.requireActual('object-hash');
  const mockWorkResultETag = jest.requireActual('__test__/test-utils/mockWorkResultETag');

  const mockWorkRequestETag = (ETagParams) => `${ETagParams.body.nGenes}-marker-genes`;
  const mockGeneExpressionETag = (ETagParams) => `${ETagParams.missingGenesBody.genes.join('-')}-expression`;

  return mockWorkResultETag(objectHash, mockWorkRequestETag, mockGeneExpressionETag);
});

jest.mock('utils/work/seekWorkResponse', () => ({
  __esModule: true,
  dispatchWorkRequest: jest.fn(() => true),
  seekFromS3: jest.fn(),
}));

let vitesscePropsSpy = null;
jest.mock('next/dynamic', () => () => (props) => {
  vitesscePropsSpy = props;
  return 'Sup Im a heatmap';
});

jest.mock('lodash/sampleSize', () => ({
  default: (collection, size) => collection.slice(0, size),
  __esModule: true,
}));

enableFetchMocks();

const mockWorkerResponses = {
  '5-marker-genes': () => Promise.resolve(_.cloneDeep(markerGenesData5)),
  '2-marker-genes': () => Promise.resolve(_.cloneDeep(markerGenesData2)),
};

const loadAndRenderDefaultHeatmap = async (storeState) => {
  await act(async () => {
    render(
      <Provider store={storeState}>
        <HeatmapPlot
          experimentId={experimentId}
          width={50}
          height={50}
        />
      </Provider>,
    );
  });
};

const mockAPIResponses = generateDefaultMockAPIResponses(experimentId);

const errorResponse = () => Promise.reject(new Error('Some error idk'));

let storeState = null;

describe('HeatmapPlot', () => {
  beforeAll(async () => {
    await preloadAll();
  });

  beforeEach(async () => {
    jest.clearAllMocks();

    fetchMock.resetMocks();
    fetchMock.doMock();

    fetchMock.mockIf(/.*/, mockAPI(mockAPIResponses));

    seekFromS3
      .mockReset()
      .mockImplementationOnce(() => null)
      .mockImplementationOnce((mockEtag) => mockWorkerResponses[mockEtag]());

    vitesscePropsSpy = null;

    storeState = makeStore();

    await storeState.dispatch(loadProcessingSettings(experimentId));
    await storeState.dispatch(loadBackendStatus(experimentId));
  });

  it('Renders the heatmap component by default if everything loads', async () => {
    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();

    // Renders correct cells and genes
    expect(vitesscePropsSpy.expressionMatrix.rows).toMatchSnapshot();
    expect(vitesscePropsSpy.expressionMatrix.cols).toMatchSnapshot();
  });

  it('Shows loader message if cellSets are loading', async () => {
    const mockLoadingAPIResponses = {
      ...mockWorkerResponses,
      [`experiments/${experimentId}/cellSets`]: () => delayedResponse({ body: 'Not found', status: 404 }, 4000),
    };

    fetchMock.mockIf(/.*/, mockAPI(mockLoadingAPIResponses));

    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    expect(screen.getByText(/We're getting your data .../i)).toBeInTheDocument();
  });

  it('Shows loader message if the marker genes are loading', async () => {
    const customWorkerResponses = {
      ...mockWorkerResponses,
      '5-marker-genes': () => delayedResponse({ body: 'Not found', status: 404 }, 4000),
    };

    await storeState.dispatch(loadCellSets(experimentId));

    seekFromS3
      .mockReset()
      .mockImplementationOnce(() => null)
      .mockImplementationOnce((mockEtag) => customWorkerResponses[mockEtag]());

    await loadAndRenderDefaultHeatmap(storeState);

    expect(screen.getByText(/We're getting your data .../i)).toBeInTheDocument();
  });

  it('Shows loader message if the marker genes are loaded but there\'s other selected genes still loading', async () => {
    const customWorkerResponses = {
      ...mockWorkerResponses,
      'loading_gene_id-expression': () => delayedResponse({ body: 'Not found', status: 404 }, 4000),
    };

    seekFromS3
      .mockReset()
      // 1st set of marker gene calls
      .mockImplementationOnce((mockEtag) => customWorkerResponses[mockEtag]())
      // 2nd set of marker gene calls
      .mockImplementationOnce((mockEtag) => customWorkerResponses[mockEtag]());

    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    // Renders correctly
    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();

    // A new gene is being loaded
    await act(async () => {
      storeState.dispatch(loadGeneExpression(experimentId, [...markerGenesData5.orderedGeneNames, 'loading_gene_id'], 'interactiveHeatmap', true));
    });

    // Loading screen shows up
    expect(screen.getByText(/We're getting your data .../i)).toBeInTheDocument();
  });

  it('Handles marker genes loading error correctly', async () => {
    const customWorkerResponses = {
      ...mockWorkerResponses,
      '5-marker-genes': errorResponse,
    };

    seekFromS3
      .mockReset()
      .mockImplementationOnce(() => Promise.resolve(null))
      .mockImplementationOnce((mockEtag) => customWorkerResponses[mockEtag]());

    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    // Error screen shows up
    expect(screen.getByText(/We're sorry, we couldn't load this./i)).toBeInTheDocument();
  });

  it('Handles expression data loading error correctly', async () => {
    const customWorkerResponses = {
      ...mockWorkerResponses,
      'loading_gene_id-expression': errorResponse,
    };

    dispatchWorkRequest
      .mockImplementationOnce(() => true)
      .mockImplementationOnce(errorResponse);

    seekFromS3
      .mockReset()
      // 1st set of marker gene calls
      .mockImplementationOnce(() => Promise.resolve(null))
      .mockImplementationOnce((mockEtag) => customWorkerResponses[mockEtag]())
      // 2nd set of marker gene calls
      .mockImplementationOnce(() => Promise.resolve(null))
      .mockImplementationOnce((mockEtag) => customWorkerResponses[mockEtag]());

    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    // Renders correctly
    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();

    // A new gene is being loaded
    await act(async () => {
      await storeState.dispatch(loadGeneExpression(experimentId, [...markerGenesData5.orderedGeneNames, 'loading_gene_id'], 'interactiveHeatmap'));
    });

    // Error screen shows up
    expect(screen.getByText(/We're sorry, we couldn't load this./i)).toBeInTheDocument();
  });

  it('Does not display hidden cell sets', async () => {
    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    // Renders correctly
    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();

    // Cell ids stored in expression matrix is string,
    // whereas cell ids stored in the store are number
    // So we need to convert them to string to be able to compare the values
    const cellsInLouvain3 = cellSetsData
      .cellSets.find(({ key }) => key === 'louvain')
      .children.find(({ name }) => name === 'Cluster 3')
      .cellIds.map((cellId) => cellId.toString());

    // It loaded once the marker genes
    expect(dispatchWorkRequest).toHaveBeenCalledTimes(1);
    expect(dispatchWorkRequest.mock.calls[0][1].name === 'MarkerHeatmap').toBe(true);

    // It shows cells in louvain-3
    expect(isSubset(cellsInLouvain3, vitesscePropsSpy.expressionMatrix.rows)).toEqual(true);

    // If a louvain-3 is suddenly hidden
    await act(async () => {
      await storeState.dispatch(setCellSetHiddenStatus('louvain-3'));
    });

    // It performs the request with the new hidden cell sets array
    expect(dispatchWorkRequest).toHaveBeenCalledTimes(2);
    expect(dispatchWorkRequest.mock.calls[1]).toMatchSnapshot();
  });

  it('Shows an empty message when all cell sets are hidden ', async () => {
    seekFromS3.mockReset();
    seekFromS3.mockImplementationOnce((mockEtag) => mockWorkerResponses[mockEtag]());

    // Last call (all the cellSets are hidden) throw the error
    seekFromS3.mockImplementationOnce(() => Promise.reject(new Error('No cells found')));

    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    // Renders correctly
    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();

    // If all cell sets are hidden
    const louvainClusterKeys = cellSetsData
      .cellSets.find(({ key: parentKey }) => parentKey === 'louvain')
      .children.map(({ key: cellSetKey }) => cellSetKey);

    const hideAllCellsPromise = louvainClusterKeys.map(async (cellSetKey) => {
      storeState.dispatch(setCellSetHiddenStatus(cellSetKey));
    });

    await act(async () => {
      await Promise.all(hideAllCellsPromise);
    });

    // The plots shows an empty message
    expect(screen.getByText(/Unhide some cell sets to show the heatmap/i)).toBeInTheDocument();
  });

  it('Reacts to cellClass groupby being changed', async () => {
    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    // Renders correctly
    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();

    // If groupedTracks change
    await act(async () => {
      await storeState.dispatch(
        updatePlotConfig('interactiveHeatmap', {
          groupedTracks: ['sample', 'louvain'],
        }),
      );
    });

    // It doesn't reorder the genes
    expect(vitesscePropsSpy.expressionMatrix.rows).toMatchSnapshot();
    // It reorders correctly
    expect(vitesscePropsSpy.expressionMatrix.cols).toMatchSnapshot();
  });

  it('Responds correctly to vitessce Heatmap callbacks', async () => {
    await storeState.dispatch(loadCellSets(experimentId));

    await loadAndRenderDefaultHeatmap(storeState);

    expect(screen.getByText(/Sup Im a heatmap/i)).toBeInTheDocument();

    // Renders the correct genes and cells
    expect(vitesscePropsSpy.expressionMatrix.rows).toMatchSnapshot();
    expect(vitesscePropsSpy.expressionMatrix.cols).toMatchSnapshot();

    // On changing the view state
    const updatedViewState = { zoom: 15, target: [1, 1] };
    await act(async () => {
      await vitesscePropsSpy.setViewState(updatedViewState);
    });

    // The viewState vitessce receives is updated
    expect(vitesscePropsSpy.viewState).toEqual(updatedViewState);

    // On hovering somewhere inside the heatmap
    const highlightedCell = '2';
    const highlightedGene = 'S100a4';
    await act(async () => {
      await vitesscePropsSpy.setCellHighlight(highlightedCell);
      await vitesscePropsSpy.setGeneHighlight(highlightedGene);
    });

    // It shows the cell info tooltip
    expect(screen.getByText(/Cell id: 2/i)).toBeInTheDocument();
    expect(screen.getByText(/Gene name: S100a4/i)).toBeInTheDocument();

    // On hovering outside
    await act(async () => {
      await vitesscePropsSpy.setCellHighlight(null);
      await vitesscePropsSpy.setGeneHighlight(null);
    });

    // It hides the tooltip
    expect(screen.queryByText(/Cell id:/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/Gene name:/i)).not.toBeInTheDocument();

    // On hovering over heatmap tracks
    await act(async () => {
      await vitesscePropsSpy.setTrackHighlight(['4', 0, 1, 2]);
    });

    // It shows the track cell info tooltip
    expect(screen.getByText(/Cell id: 4/i)).toBeInTheDocument();
    expect(screen.getByText(/Group name: Cluster 0/i)).toBeInTheDocument();

    // On hovering outside heatmap tracks
    await act(async () => {
      await vitesscePropsSpy.setTrackHighlight(null);
    });

    // It hides the track cell info tooltip
    expect(screen.queryByText(/Cell id:/i)).not.toBeInTheDocument();
    expect(screen.queryByText(/Group name:/i)).not.toBeInTheDocument();

    // And doesn't show the normal cell info again
    expect(screen.queryByText(/Gene name:/i)).not.toBeInTheDocument();
  });
});
