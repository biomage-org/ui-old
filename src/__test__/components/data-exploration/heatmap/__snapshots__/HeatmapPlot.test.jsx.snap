// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeatmapPlot Does not display hidden cell sets 1`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "16",
  "17",
  "18",
  "19",
  "20",
  "21",
  "22",
  "23",
  "24",
  "25",
  "26",
  "27",
  "28",
  "29",
  "30",
  "31",
  "32",
  "33",
  "34",
  "35",
  "36",
  "37",
  "38",
  "39",
  "40",
  "41",
  "42",
  "43",
  "47",
  "48",
  "49",
  "50",
  "51",
  "52",
  "53",
  "54",
  "55",
  "57",
  "58",
  "59",
  "60",
  "66",
  "68",
  "69",
  "70",
  "71",
  "72",
  "73",
  "75",
  "76",
  "77",
  "79",
  "80",
  "81",
  "82",
  "84",
  "85",
  "88",
  "89",
  "90",
  "91",
  "92",
  "93",
  "94",
  "95",
  "96",
  "97",
  "98",
  "99",
  "100",
]
`;

exports[`HeatmapPlot Does not display hidden cell sets 2`] = `
Array [
  "Ms4a4b",
  "Smc4",
  "Ccr7",
  "Ifi27l2a",
  "Gm8369",
  "S100a4",
  "S100a6",
  "Tmem176a",
  "Tmem176b",
  "Cxcr6",
  "5830411N06Rik",
  "Lmo4",
  "Il18r1",
  "Atp2b1",
  "Pde5a",
  "Ccl5",
  "Nkg7",
  "Klrd1",
  "AW112010",
  "Klrc1",
  "Gzma",
  "Stmn1",
  "Hmgn2",
  "Pclaf",
  "Tuba1b",
  "Lyz2",
  "Ifitm3",
  "Fcer1g",
  "Tyrobp",
  "Cst3",
  "Cd74",
  "Igkc",
  "Cd79a",
  "H2-Ab1",
  "H2-Eb1",
]
`;

exports[`HeatmapPlot Does not display hidden cell sets 3`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "16",
  "17",
  "18",
  "19",
  "20",
  "21",
  "22",
  "23",
  "24",
  "25",
  "26",
  "27",
  "28",
  "29",
  "30",
  "31",
  "32",
  "33",
  "34",
  "35",
  "36",
  "37",
  "38",
  "39",
  "40",
  "41",
  "42",
  "43",
  "47",
  "48",
  "49",
  "50",
  "51",
  "52",
  "53",
  "54",
  "55",
  "57",
  "58",
  "59",
  "60",
  "61",
  "62",
  "64",
  "65",
  "66",
  "68",
  "69",
  "70",
  "71",
  "72",
  "73",
  "75",
  "76",
  "77",
  "79",
  "80",
  "81",
  "82",
  "84",
  "85",
  "88",
  "89",
  "90",
  "91",
  "92",
  "93",
  "94",
  "95",
  "96",
  "97",
  "98",
  "99",
  "100",
]
`;

exports[`HeatmapPlot Does not display hidden cell sets 4`] = `
Array [
  "Ms4a4b",
  "Smc4",
  "Ccr7",
  "Ifi27l2a",
  "Gm8369",
  "S100a4",
  "S100a6",
  "Tmem176a",
  "Tmem176b",
  "Cxcr6",
  "5830411N06Rik",
  "Lmo4",
  "Il18r1",
  "Atp2b1",
  "Pde5a",
  "Ccl5",
  "Nkg7",
  "Klrd1",
  "AW112010",
  "Klrc1",
  "Gzma",
  "Stmn1",
  "Hmgn2",
  "Pclaf",
  "Tuba1b",
  "Lyz2",
  "Ifitm3",
  "Fcer1g",
  "Tyrobp",
  "Cst3",
  "Cd74",
  "Igkc",
  "Cd79a",
  "H2-Ab1",
  "H2-Eb1",
]
`;

exports[`HeatmapPlot Reacts to cellClass groupby being changed 1`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "16",
  "17",
  "18",
  "19",
  "20",
  "21",
  "22",
  "23",
  "24",
  "25",
  "26",
  "27",
  "28",
  "29",
  "30",
  "31",
  "32",
  "33",
  "34",
  "35",
  "36",
  "37",
  "38",
  "39",
  "40",
  "41",
  "42",
  "43",
  "47",
  "48",
  "49",
  "50",
  "51",
  "52",
  "53",
  "54",
  "55",
  "57",
  "58",
  "59",
  "60",
  "61",
  "62",
  "64",
  "65",
  "66",
  "68",
  "69",
  "70",
  "71",
  "72",
  "73",
  "75",
  "76",
  "77",
  "79",
  "80",
  "81",
  "82",
  "84",
  "85",
  "88",
  "89",
  "90",
  "91",
  "92",
  "93",
  "94",
  "95",
  "96",
  "97",
  "98",
  "99",
  "100",
]
`;

exports[`HeatmapPlot Reacts to cellClass groupby being changed 2`] = `
Array [
  "Ms4a4b",
  "Smc4",
  "Ccr7",
  "Ifi27l2a",
  "Gm8369",
  "S100a4",
  "S100a6",
  "Tmem176a",
  "Tmem176b",
  "Cxcr6",
  "5830411N06Rik",
  "Lmo4",
  "Il18r1",
  "Atp2b1",
  "Pde5a",
  "Ccl5",
  "Nkg7",
  "Klrd1",
  "AW112010",
  "Klrc1",
  "Gzma",
  "Stmn1",
  "Hmgn2",
  "Pclaf",
  "Tuba1b",
  "Lyz2",
  "Ifitm3",
  "Fcer1g",
  "Tyrobp",
  "Cst3",
  "Cd74",
  "Igkc",
  "Cd79a",
  "H2-Ab1",
  "H2-Eb1",
]
`;

exports[`HeatmapPlot Renders the heatmap component by default if everything loads 1`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "16",
  "17",
  "18",
  "19",
  "20",
  "21",
  "22",
  "23",
  "24",
  "25",
  "26",
  "27",
  "28",
  "29",
  "30",
  "31",
  "32",
  "33",
  "34",
  "35",
  "36",
  "37",
  "38",
  "39",
  "40",
  "41",
  "42",
  "43",
  "47",
  "48",
  "49",
  "50",
  "51",
  "52",
  "53",
  "54",
  "55",
  "57",
  "58",
  "59",
  "60",
  "61",
  "62",
  "64",
  "65",
  "66",
  "68",
  "69",
  "70",
  "71",
  "72",
  "73",
  "75",
  "76",
  "77",
  "79",
  "80",
  "81",
  "82",
  "84",
  "85",
  "88",
  "89",
  "90",
  "91",
  "92",
  "93",
  "94",
  "95",
  "96",
  "97",
  "98",
  "99",
  "100",
]
`;

exports[`HeatmapPlot Renders the heatmap component by default if everything loads 2`] = `
Array [
  "Ms4a4b",
  "Smc4",
  "Ccr7",
  "Ifi27l2a",
  "Gm8369",
  "S100a4",
  "S100a6",
  "Tmem176a",
  "Tmem176b",
  "Cxcr6",
  "5830411N06Rik",
  "Lmo4",
  "Il18r1",
  "Atp2b1",
  "Pde5a",
  "Ccl5",
  "Nkg7",
  "Klrd1",
  "AW112010",
  "Klrc1",
  "Gzma",
  "Stmn1",
  "Hmgn2",
  "Pclaf",
  "Tuba1b",
  "Lyz2",
  "Ifitm3",
  "Fcer1g",
  "Tyrobp",
  "Cst3",
  "Cd74",
  "Igkc",
  "Cd79a",
  "H2-Ab1",
  "H2-Eb1",
]
`;

exports[`HeatmapPlot Responds correctly to vitessce Heatmap callbacks 1`] = `
Array [
  "1",
  "2",
  "3",
  "4",
  "5",
  "6",
  "7",
  "8",
  "9",
  "10",
  "11",
  "12",
  "16",
  "17",
  "18",
  "19",
  "20",
  "21",
  "22",
  "23",
  "24",
  "25",
  "26",
  "27",
  "28",
  "29",
  "30",
  "31",
  "32",
  "33",
  "34",
  "35",
  "36",
  "37",
  "38",
  "39",
  "40",
  "41",
  "42",
  "43",
  "47",
  "48",
  "49",
  "50",
  "51",
  "52",
  "53",
  "54",
  "55",
  "57",
  "58",
  "59",
  "60",
  "61",
  "62",
  "64",
  "65",
  "66",
  "68",
  "69",
  "70",
  "71",
  "72",
  "73",
  "75",
  "76",
  "77",
  "79",
  "80",
  "81",
  "82",
  "84",
  "85",
  "88",
  "89",
  "90",
  "91",
  "92",
  "93",
  "94",
  "95",
  "96",
  "97",
  "98",
  "99",
  "100",
]
`;

exports[`HeatmapPlot Responds correctly to vitessce Heatmap callbacks 2`] = `
Array [
  "Ms4a4b",
  "Smc4",
  "Ccr7",
  "Ifi27l2a",
  "Gm8369",
  "S100a4",
  "S100a6",
  "Tmem176a",
  "Tmem176b",
  "Cxcr6",
  "5830411N06Rik",
  "Lmo4",
  "Il18r1",
  "Atp2b1",
  "Pde5a",
  "Ccl5",
  "Nkg7",
  "Klrd1",
  "AW112010",
  "Klrc1",
  "Gzma",
  "Stmn1",
  "Hmgn2",
  "Pclaf",
  "Tuba1b",
  "Lyz2",
  "Ifitm3",
  "Fcer1g",
  "Tyrobp",
  "Cst3",
  "Cd74",
  "Igkc",
  "Cd79a",
  "H2-Ab1",
  "H2-Eb1",
]
`;
